src/dms/clis/EdFi.DataManagementService.ApiSchemaDownloader/Services/ApiSchemaDownloader.cs

```diff
@@ -0,0 +1,196 @@
+using NuGet.Configuration;
+using NuGet.Packaging;
+using NuGet.Packaging.Core;
+using NuGet.Protocol.Core.Types;
+using NuGet.Versioning;
+
+namespace EdFi.DataManagementService.ApiSchemaDownloader.Services
+{
+    public class ApiSchemaDownloader(ILogger<ApiSchemaDownloader> logger) : IApiSchemaDownloader
+    {
+        private readonly ILogger<ApiSchemaDownloader> _logger =
+            logger ?? throw new ArgumentNullException(nameof(logger));
+
+        public void ExtractApiSchemaJsonFromAssembly(string packageId, string packagePath, string outputDir)
+        {
+            _logger.LogInformation(
+                "Extracting API schema from package {PackageId} located at {PackagePath}",
+                packageId,
+                packagePath
+            );
+
+            try
+            {
+                using var packageReader = new PackageArchiveReader(packagePath);
+                var dllPath = packageReader.GetFiles().FirstOrDefault(f => f.EndsWith("ApiSchema.dll"));
+                if (dllPath == null)
+                {
+                    _logger.LogError("No DLL found in the package {PackageId}", packageId);
+                    throw new Exception("No DLL found in the package.");
+                }
+
+                var tempDir = Path.Combine(Path.GetTempPath(), Guid.NewGuid().ToString());
+                Directory.CreateDirectory(tempDir);
+                var dllFilePath = Path.Combine(tempDir, Path.GetFileName(dllPath));
+
+                _logger.LogDebug("Extracting DLL {DllPath} to {TempDir}", dllPath, tempDir);
+
+                using (var dllStream = packageReader.GetStream(dllPath))
+                using (var fileStream = File.Create(dllFilePath))
+                {
+                    dllStream.CopyTo(fileStream);
+                }
+
+                _logger.LogDebug("Loading assembly from {DllFilePath}", dllFilePath);
+                byte[] assemblyBytes = File.ReadAllBytes(dllFilePath);
+                var assembly = Assembly.Load(assemblyBytes);
+
+                var resourceNames = assembly
+                    .GetManifestResourceNames()
+                    .Where(name => name.StartsWith(packageId, StringComparison.OrdinalIgnoreCase))
+                    .ToList();
+
+                if (!resourceNames.Any())
+                {
+                    _logger.LogError(
+                        "ApiSchema.json not found as an embedded resource in {PackageId}",
+                        packageId
+                    );
+                    throw new Exception("ApiSchema.json not found as an embedded resource.");
+                }
+
+                _logger.LogInformation(
+                    "Extracting {ResourceCount} resources from assembly.",
+                    resourceNames.Count
+                );
+
+                foreach (var resourceName in resourceNames)
+                {
+                    string outputFilePath = Path.Combine(outputDir, resourceName.Replace("{packageId}.", ""));
```

Comment:
I guess the intention here is to remove packageId from the file name like: ${packageId}.